### ISSUE TITLE
right click menu option 'Fast-Forward to branch' doesn't work

### ISSUE THREAD

**Issue by zoffyzhang** [ForkIssues/TrackerWin#700]
Fork: 1.46.1.0

Hi, I just get started to use Fork, great git GUI by the way.

If I merged a feature branch to develop branch and select "No Fast-Forward", then I can't fast-forward the feature branch to the develop branch, I guess it's a bug.

<img width="369" alt="微信截图_20200305153434" src="https://user-images.githubusercontent.com/11470090/75964827-d8cb5900-5f02-11ea-93bb-b3d16734954f.png">

<img width="247" alt="Snipaste_2020-03-05_15-46-49" src="https://user-images.githubusercontent.com/11470090/75964872-e7b20b80-5f02-11ea-8844-4eaa42bac5e6.png">

**Comment by DanPristupov** [comment]
Fast-forward basically means 'move the branch pointer, don't create a merge commit'. It's only possible when both branches are on the same line.

So, when you create a merge commit (or any commit) in your local branch, you can not fast-forward anymore.

**Comment by zoffyzhang** [comment]
Moving the branch pointer is exactly what I want, so that I can get other people's commit into my branch, this is a valid operation in command line. 

When a branch merge into another branch using --no-ff, no matter they are conflicted or not, once the merge is done, they are OK to fast-forward.

**Comment by DanPristupov** [comment]
Can you show a screenshot of the graph please?

**Comment by zoffyzhang** [comment]
<img width="41" alt="微信截图_20200305180012" src="https://user-images.githubusercontent.com/11470090/75970363-2e0b6880-5f0b-11ea-9eac-e8f082ecb112.png">

**Comment by DanPristupov** [comment]
Can you show one where develop and the feature branch are visible?

**Comment by zoffyzhang** [comment]
<img width="448" alt="微信截图_20200305180353" src="https://user-images.githubusercontent.com/11470090/75970688-b1c55500-5f0b-11ea-8b10-5e42237c2775.png">

**Comment by DanPristupov** [comment]
Well, I think you must be able to fast-forward develop -> origin/develop in your case.

A similar case (update br1 -> origin/br1)

Before:
<img width="108" alt="Screenshot 2020-03-05 at 11 15 58" src="https://user-images.githubusercontent.com/618115/75971880-b28dc500-5ed2-11ea-9826-25160f64c90d.png">

Fast-forward br1:

<img width="125" alt="Screenshot 2020-03-05 at 11 16 47" src="https://user-images.githubusercontent.com/618115/75971949-cf29fd00-5ed2-11ea-8524-661804d23947.png">

br1 is up-to-date now.

**Comment by zoffyzhang** [comment]
in your screenshot, it looks like you performed follow commands

```git
1. git reset --hard {SHA}

2. git merge origin/br1 --ff 
or git pull --ff
```

I think I know what went wrong, it's me that misunderstood the Fork feature. Take the following screenshot as an example:

![Snipaste_2020-03-05_20-47-55](https://user-images.githubusercontent.com/11470090/75983119-dd076e80-5f22-11ea-9d1f-072268a03271.png)

the dev is one commit behind master, and the current branch is dev.

so now, I thought the "Fast-Forward to origin/master" option will perform command `git merge master --ff-only`, but it actually perform something like
```git
git checkout master 
git merge origin/master --ff-only 
git checkout dev
```

my bad.

**Comment by DanPristupov** [comment]
> but it actually perform something like ...

Yes, this is absolutely correct 👍!

The point is you don't need to checkout a branch when you want to make other branch up to date.
